content/purgatorio/now-go-build/transfer-learning.md

```diff
@@ -1,5 +1,111 @@
 ---
 title: Transfer Learning
 author: clone95
-description: ---
----
\ No newline at end of file
+description: A detailed guide about what is Transfer Learning, how to use it, and a lot of resources found on the web that shares pre-trained models.
+---
+
+# Index 
+
+- [What is Transfer Learning](#Why-Do-You-Need-Transfe-Learning)
+- [Transfer Learning Magical Properties](#[TransferLearning-Inner-Workings)
+- [How Does Transfer Learning Work](#How-Does-Transfer-Learning-Work)
+- [Where to Find Pretrained Models](#Where-to-Find-Pretrained-Models)
+- [Conclusions](#Conclusions)
+
+##  What is Transfer Learning
+
+[Transfer Learning](https://en.wikipedia.org/wiki/Transfer_learning) is a research problem in Machine Learning that focuses on storing knowledge gained while solving one problem and applying it to a different but related problem. 
+
+This is done by _pre-training_ a model on a dataset (usually bigger and with generic content, like [ImageNet](http://www.image-net.org/)), and then _fine-tuning_ the model on a smaller and similar dataset.
+
+For example, you can fine-tune a pre-trained model from ImageNet on your much smaller dataset of images, in order to benefit the pre-training and obtain a series of almost-magical properties and results.
+
+To get a comprehensive overview of Why, What and How is used Transfer Learning, plus possible applications, read this awesome article:
+
+- [Transfer Learning - Machine Learning's Next Frontier](https://ruder.io/transfer-learning/index.html)
+
+
+#### Transfer Learning Magical Properties
+
+The theory of Transfer Learning suggests that when using a pre-trained model on a similar dataset:
+
+**You need less**:
+- Data: you need much smaller datasets to obtain the same (or better) model performances
+- Computational Power / Computational Time: empirical results have shown that pre-trained models reach top performances after much fewer epochs of training, that starting the training from scratch
+
+**You improve:**
+- Out-of-sample generalization: when predicting on test data you often obtain better accuracy, thus reducing overfitting (or whatever metric you're using)
+- Robustness: you make the model more "robust" to real-world low-quality data  
+
+So, **the first thing to do** every time you frame a new problem, is to ask yourself:
+
+**Can I leverage Transfer Learning in solving this problem?**
+
+
+## How Does Transfer Learning Work
+
+The practice of Transfer Learning allows reusing most of the parameters (weights) of a neural network previously trained on a problem similar to the one we have to solve, dwelling on the training only of the last layers that are usually those dedicated to the classification and/or regression of the features obtained with the previous layers.
```

Comment:
> of a neural network
Only neural network trained models can be used for transfer-learning? Is this true? And why so? What about other models that are not NN-based? Why not?

Please check this, its super important, or else incorrect generalisation can be assumed from it.

Reply:
> dwelling on the training only of the last layers that are usually those dedicated to the classification and/or regression of the features obtained with the previous layers.

I think there might a quick jump into explaining the mechanics of TL, instead a separate heading could be used to explain how it works or how it is done?

Reply:
As there are multiple types of this so-called weights freezing (name the high-level names when explaining at least one of them).

Reply:
nice comments, looking forward to having a little help from you expanding these points

Reply:
Ok need to search for this, and will get back to you

Reply:
Have a look at this article: https://machinelearningmastery.com/transfer-learning-for-deep-learning/ and here as well https://www.datacamp.com/community/tutorials/transfer-learning

Explains how TL can be applied to CV and NLP - so this needs to be made clear

Reply:
Types of freezing will share the related resources.